### ISSUE TITLE
Evalhook 0.5.6, multiple line function definition no longer works.

### ISSUE THREAD

**Issue by huxoll** [tario/evalhook#7]
I updated to the new gem version (with related dependencies) today.

The following code no longer works:

``` ruby
require "rubygems"
require "shikashi"

include Shikashi

module Sandboxer
  extend self
end

sandbox = Sandbox.new
priv = Privileges.new
priv.allow_method :puts

priv.instances_of(Sandboxer).allow_all

sandbox.run(priv, '
def go()
  puts "Printing a line!"
  puts "Printing a line!"
end', :base_namespace => Sandboxer)

Sandboxer.method("go").call()
```

The output only shows "Printing a line!" once (note that the go method makes 2 calls to puts.)

However, using the old dependencies,
"partialruby", '= 0.2.3'
"ruby_parser", '= 2.3.1'
"evalhook", '= 0.5.5'

The output shows: 
Printing a line!
Printing a line!
as expected.

**Comment by tario** [comment]
If you need a quick patch for this, replace tree_processor.rb, line 220 
this:

``` ruby
      s(tree[0],tree[1],tree[2],process(tree[3]))
```

with this:

``` ruby
      s(tree[0],tree[1],tree[2],*tree[3..-1].map(&method(:process)))
```

(Or keep using the old version)

**Comment by tario** [comment]
Fixed on released evalhook v0.5.7
